docs: Cluster Peering beta (#14011)
hashicorp/consul: website/content/docs/connect/cluster-peering/create-manage-peering.mdx

# Create and Manage Peering Connections

~> **Cluster peering is currently in beta:** Functionality associated with cluster peering is subject to change. You should never use the beta release in secure environments or production scenarios. Features in beta may have performance issues, scaling issues, and limited support.

A peering token enables cluster peering between different datacenters. Once you generate a peering token, you can use it to establish a connection between clusters. Then you can export services and create intentions so that peered clusters can call those services.

> ## Create a peering connection

Reviewer: There are some new security features implemented last week that need to be accounted for here and in the API docs for peering:
* Every time a peering token is generated there is also a one-time-use establishment secret that is created and embedded in the token. Re-generating a peering token will invalidate the last establishment secret. Meaning that to establish a peering the latest token must be used.
* `/peering/establish` can only be dialed once for a given peering token. After establishing the peering a peering token cannot be re-used and a new one must be generated to re-establish the peering (if needed).

# Create and Manage Peering Connections

~> **Cluster peering is currently in beta:** Functionality associated with cluster peering is subject to change. You should never use the beta release in secure environments or production scenarios. Features in beta may have performance issues, scaling issues, and limited support.

A peering token enables cluster peering between different datacenters. Once you generate a peering token, you can use it to establish a connection between clusters. Then you can export services and create intentions so that peered clusters can call those services.

## Create a peering connection

Cluster peering is not enabled by default on Consul servers. To peer clusters, you must first configure all Consul servers so that `peering` is `enabled`. For additional information, refer to [Configuration Files](/docs/agent/config/config-files).

> Then, complete the following steps in order:

Reviewer 2: ```suggestion
After enabling peering for all Consul servers, complete the following steps in order:
```
For me, starting the paragraph with "Then" has the effect of making the text feel like a step is missing. You might consider making step 1 "Configure all Consul servers to enable so that `peering` is `enabled`."

1. Create a peering token
1. Establish a connection between clusters
1. Export services between clusters
1. Authorize services for peers

You can generate peering tokens and initiate connections on any available agent using either the API or the Consul UI. If you use the API, we recommend performing these operations through a client agent in the partition you want to connect.

The UI does not currently support exporting services between clusters or authorizing services for peers.

### Create a peering token

To begin the cluster peering process, generate a peering token in one of your clusters. The other cluster uses this token to establish the peering connection.

> Everytime you generate a peering token, a single-use establishment secret is embedded in the token. Because regenerating a peering token invalidates the previously generated secret, you must use the most recently created token to establish peering connections.

Reviewer 2: ```suggestion
Every time you generate a peering token, a single-use establishment secret is embedded in the token. Because regenerating a peering token invalidates the previously generated secret, you must use the most recently created token to establish peering connections.
```
typo

$ consul config write peering-intentions.hcl
```

## Manage peering connections

> After you establish a peering connection, you can get a list of all active peering connections, read a specific peering connection's info, check peering connection health, and delete peering connections.

Reviewer 2: ```suggestion
After you establish a peering connection, you can get a list of all active peering connections, read a specific peering connection's information, check peering connection health, and delete peering connections.
```
Unless we mean `INFO` logging message, we should probably spell this out.

### List all peering connections

You can list all active peering connections in a cluster.

<Tabs>
<Tab heading="Consul API">

> After you establish a peering connection, [query the `/peering/` endpoint](/api-docs/peering#list-all-peerings) to get a list of all peering connections. For example, the following command requests a list of all peering connections on `localhost` and returns the info as a series of JSON objects:

Reviewer 2: ```suggestion
After you establish a peering connection, [query the `/peering/` endpoint](/api-docs/peering#list-all-peerings) to get a list of all peering connections. For example, the following command requests a list of all peering connections on `localhost` and returns the information as a series of JSON objects:
```

```shell-session
$  curl http://127.0.0.1:8500/v1/peerings

[
    {
        "ID": "462c45e8-018e-f19d-85eb-1fc1bcc2ef12",
        "Name": "cluster-02",
        "State": "ACTIVE",
        "Partition": "default",
        "PeerID": "e83a315c-027e-bcb1-7c0c-a46650904a05",
        "PeerServerName": "server.dc1.consul",
        "PeerServerAddresses": [
            "10.0.0.1:8300"
        ],
        "CreateIndex": 89,
        "ModifyIndex": 89
    },
    {
        "ID": "1460ada9-26d2-f30d-3359-2968aa7dc47d",
        "Name": "cluster-03",
        "State": "INITIAL",
        "Partition": "default",
        "Meta": {
            "env": "production"
        },
        "CreateIndex": 109,
        "ModifyIndex": 119
    },
]
```
</Tab>

<Tab heading="Consul UI">

In the Consul UI, click **Peers**. The UI lists peering connections you created for clusters in a datacenter.

The name that appears in the list is the name of the cluster in a different datacenter with an established peering connection.
</Tab>
</Tabs>

### Read a peering connection

You can get information about individual peering connections between clusters.

<Tabs>
<Tab heading="Consul API">

> After you establish a peering connection, [query the `/peering/:name` endpoint](/api-docs/peering#read-a-peering-connection) to get peering information about for a specific cluster. For example, the following command requests peering connection info for "cluster-02" and returns the info as a JSON object:

Reviewer 2: ```suggestion
After you establish a peering connection, [query the `/peering/:name` endpoint](/api-docs/peering#read-a-peering-connection) to get peering information about for a specific cluster. For example, the following command requests peering connection information for "cluster-02" and returns the info as a JSON object:
```
